backend/es/views.py

```diff
@@ -62,4 +68,22 @@ def post(self, request, *args, **kwargs):
             logger.exception(e)
             raise APIException('Search failed')
 
+        if history_obj and results:
+            self._log_query_done(history_obj, results)
+
         return Response(results)
+
+    def _log_query_started(self, request, corpus_name, es_query):
+        if save_query(request.user, es_query):
+            corpus = Corpus.objects.get(name=corpus_name)
+            return Query.objects.create(
+                user=request.user,
+                corpus=corpus,
+                query_json=es_query,
+            )
+
+    def _log_query_done(self, query, results):
```

Comment:
These functions don't log, but rather create/save, would change the name to indicate this.